admin/helpers/routes.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose')
+
+const School = require('../models/school')
+const Pupil = require('../models/pupil')
+const Answer = require('../models/answer')
+
+const fetchPupilsData = async (req, next) => {
```

Comment:
* This function has dependencies on the session (under `req`) for the school associated with the user, and `next`.  In my view it would be cleaner if this function took only parameter - the school id and threw an error rather than called `next()`.  I don't think it is the responsibility of the helper function to decide what to do.

* The name of this function is `fetchPupilsData` but it returns pupils mongoose objects and school data.  This mix is inconsistent - as one is a plain javascript object, and one is an array of mongoose objects.

* This function does two fetches in series, but they should be done in parallel as the school id is known before the first call.

* This looks like it ought to be in a service (e.g. `pupilService`) rather than a helper function and should behave appropriately (as described above).